ClubDoorman/BadMessageManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Security.Cryptography;
+using System.Text;
+
+namespace ClubDoorman;
+
+public class BadMessageManager
+{
+    private const string Path = "data/bad-messages.txt";
+    private readonly SemaphoreSlim _semaphore = new(1);
+    private readonly HashSet<string> _bad = File.ReadAllLines(Path).ToHashSet();

```

Comment:
I dislike this approach because we are:

Hashing messages to 512 bytes
Converting it to base64 that increases size by about 25%

what if we just use base64 as means to encode this to a text file

as for how byte[64] should be stored and compared in memory, I cannot say without benchmarks what would be the best. I guess just treating every byte as char and concatenating it to string is definitely good enough for out purposes

Reply:
Actually you know what, this would lead to messing up with encoding and stuff, let's stick to base64 as a fool-proof thing. Paying 25% extra memory is not such a big deal, even on 1000th of messages it's negligable.